Low API example fails as written

I'm currently reading http://werkzeug.pocoo.org/docs/0.11/levels/ and I copied the example of the low API, adding a minimal harness to make it work:
```python
from werkzeug.formparser import parse_form_data
from werkzeug.utils import escape

def hello_world(environ, start_response):
    result = ['<title>Greeter</title>']
    if environ['REQUEST_METHOD'] == 'POST':
        form = parse_form_data(environ)[1]
        result.append('<h1>Hello %s!</h1>' % escape(form['name']))
    result.append('''
        <form action="" method="post">
            <p>Name: <input type="text" name="name" size="20">
            <input type="submit" value="Greet me">
        </form>
    ''')
    start_response('200 OK', [('Content-Type', 'text/html; charset=utf-8')])
    return [''.join(result)]

if __name__ == '__main__':
    from werkzeug.serving import run_simple
    run_simple('127.0.0.1', 5000, hello_world)
```

But, when sending back any string (e.g. `world`) through the form, I get an error:
```python
Error on request:
Traceback (most recent call last):
  File "/Users/jacquerie/Code/werkzeug/werkzeug/werkzeug/serving.py", line 193, in run_wsgi
    execute(self.server.app)
  File "/Users/jacquerie/Code/werkzeug/werkzeug/werkzeug/serving.py", line 184, in execute
    write(data)
  File "/Users/jacquerie/Code/werkzeug/werkzeug/werkzeug/serving.py", line 164, in write
    assert isinstance(data, bytes), 'applications must write bytes'
AssertionError: applications must write bytes
```

I can see that this is easily fixed by forcing everything to be bytes, as in:
```python
return [''.join(result).encode('utf-8')]
```
and that this is caused by `form['name']` not being an instance of `bytes`, but I'm not sure if I should push a commit which is just a workaround to (maybe?) a deeper issue.

--- a/docs/levels.rst
+++ b/docs/levels.rst
@@ -51,7 +51,7 @@ objects but by taking advantage of the parsing functions werkzeug provides::
             </form>
         ''')
         start_response('200 OK', [('Content-Type', 'text/html; charset=utf-8')])
-        return [''.join(result)]
+        return [''.join(result).encode('utf-8')]
 
 High or Low?
 ============

